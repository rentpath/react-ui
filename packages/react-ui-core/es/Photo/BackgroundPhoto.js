import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

var _dec, _class, _class2, _temp;

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import pipe from 'lodash/fp/pipe';
import filter from 'lodash/fp/filter';
import map from 'lodash/fp/map';
import join from 'lodash/fp/join';
import flatten from 'lodash/flatten';
import themed from '@rentpath/react-themed';
var backgroundUrl = pipe(function () {
  for (var _len = arguments.length, urls = new Array(_len), _key = 0; _key < _len; _key++) {
    urls[_key] = arguments[_key];
  }

  return urls.concat();
}, flatten, filter(Boolean), // skip if undefined
map(function (url) {
  return "url(".concat(url, ")");
}), join(', '));
var BackgroundPhoto = (_dec = themed(['BackgroundPhoto'], {
  pure: true
}), _dec(_class = (_temp = _class2 =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(BackgroundPhoto, _PureComponent);

  function BackgroundPhoto() {
    _classCallCheck(this, BackgroundPhoto);

    return _possibleConstructorReturn(this, _getPrototypeOf(BackgroundPhoto).apply(this, arguments));
  }

  _createClass(BackgroundPhoto, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          theme = _this$props.theme,
          className = _this$props.className,
          url = _this$props.url,
          fallbackUrl = _this$props.fallbackUrl,
          alt = _this$props.alt,
          rest = _objectWithoutProperties(_this$props, ["theme", "className", "url", "fallbackUrl", "alt"]);

      return React.createElement("div", _extends({
        style: {
          backgroundImage: backgroundUrl(url, fallbackUrl)
        },
        className: classnames(className, theme.BackgroundPhoto),
        "data-tid": "bg-photo"
      }, rest));
    }
  }]);

  return BackgroundPhoto;
}(PureComponent), _defineProperty(_class2, "propTypes", {
  url: PropTypes.string,
  fallbackUrl: PropTypes.string,
  theme: PropTypes.object,
  className: PropTypes.string,
  alt: PropTypes.string
}), _defineProperty(_class2, "defaultProps", {
  theme: {}
}), _temp)) || _class);
export { BackgroundPhoto as default };