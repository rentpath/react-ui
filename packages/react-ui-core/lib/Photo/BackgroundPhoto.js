'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _dec, _class, _class2, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _pipe = require('lodash/fp/pipe');

var _pipe2 = _interopRequireDefault(_pipe);

var _filter = require('lodash/fp/filter');

var _filter2 = _interopRequireDefault(_filter);

var _map = require('lodash/fp/map');

var _map2 = _interopRequireDefault(_map);

var _join = require('lodash/fp/join');

var _join2 = _interopRequireDefault(_join);

var _flatten = require('lodash/flatten');

var _flatten2 = _interopRequireDefault(_flatten);

var _reactThemed = require('react-themed');

var _reactThemed2 = _interopRequireDefault(_reactThemed);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var backgroundUrl = (0, _pipe2.default)(function () {
  for (var _len = arguments.length, urls = Array(_len), _key = 0; _key < _len; _key++) {
    urls[_key] = arguments[_key];
  }

  return [].concat(urls);
}, _flatten2.default, (0, _filter2.default)(Boolean), // skip if undefined
(0, _map2.default)(function (url) {
  return 'url(' + url + ')';
}), (0, _join2.default)(', '));

var BackgroundPhoto = (_dec = (0, _reactThemed2.default)(['BackgroundPhoto'], {
  pure: true
}), _dec(_class = (_temp = _class2 = function (_PureComponent) {
  (0, _inherits3.default)(BackgroundPhoto, _PureComponent);

  function BackgroundPhoto() {
    (0, _classCallCheck3.default)(this, BackgroundPhoto);
    return (0, _possibleConstructorReturn3.default)(this, (BackgroundPhoto.__proto__ || (0, _getPrototypeOf2.default)(BackgroundPhoto)).apply(this, arguments));
  }

  (0, _createClass3.default)(BackgroundPhoto, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          theme = _props.theme,
          className = _props.className,
          url = _props.url,
          fallbackUrl = _props.fallbackUrl,
          alt = _props.alt,
          rest = (0, _objectWithoutProperties3.default)(_props, ['theme', 'className', 'url', 'fallbackUrl', 'alt']);


      return _react2.default.createElement('div', (0, _extends3.default)({
        style: {
          backgroundImage: backgroundUrl(url, fallbackUrl)
        },
        className: (0, _classnames2.default)(className, theme.BackgroundPhoto),
        'data-tid': 'bg-photo'
      }, rest));
    }
  }]);
  return BackgroundPhoto;
}(_react.PureComponent), _class2.propTypes = {
  url: _propTypes2.default.string,
  fallbackUrl: _propTypes2.default.string,
  theme: _propTypes2.default.object,
  className: _propTypes2.default.string,
  alt: _propTypes2.default.string
}, _class2.defaultProps = {
  theme: {}
}, _temp)) || _class);
exports.default = BackgroundPhoto;