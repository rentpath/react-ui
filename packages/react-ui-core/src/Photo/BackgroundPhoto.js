import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import pipe from 'lodash/fp/pipe'
import filter from 'lodash/fp/filter'
import map from 'lodash/fp/map'
import join from 'lodash/fp/join'
import flatten from 'lodash/flatten'
import themed from '@rentpath/react-themed'

const backgroundUrl = pipe(
  (...urls) => ([...urls]),
  flatten,
  filter(Boolean), // skip if undefined
  map(url => `url(${url})`),
  join(', ')
)

@themed(['BackgroundPhoto'], {
  pure: true,
})
export default class BackgroundPhoto extends PureComponent {
  static propTypes = {
    url: PropTypes.string,
    fallbackUrl: PropTypes.string,
    theme: PropTypes.object,
    className: PropTypes.string,
    alt: PropTypes.string,
  }

  static defaultProps = {
    theme: {},
  }

  render() {
    const {
      theme,
      className,
      url,
      fallbackUrl,
      alt,
      ...rest
    } = this.props

    return (
      <div
        style={{
          backgroundImage: backgroundUrl(url, fallbackUrl),
        }}
        className={classnames(
          className,
          theme.BackgroundPhoto
        )}
        data-tid="bg-photo"
        {...rest}
      />
    )
  }
}
