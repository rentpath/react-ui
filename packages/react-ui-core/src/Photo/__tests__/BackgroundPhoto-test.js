import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import BackgroundPhoto from '../BackgroundPhoto'

describe('BackgroundPhoto', () => {
  it('passes extra props through', () => {
    const wrapper = shallow(
      <BackgroundPhoto
        data-tag_foo="testing"
      />
    )

    expect(wrapper.prop('data-tag_foo')).toEqual('testing')
  })

  describe('fallbackUrl', () => {
    it('renders 1 url when not set', () => {
      const snap = renderer
        .create(<BackgroundPhoto url="/foo" />)
        .toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('renders 2 urls when set', () => {
      const snap = renderer
        .create(
          <BackgroundPhoto
            url="/foo"
            fallbackUrl="/bar"
          />
        )
        .toJSON()
      expect(snap).toMatchSnapshot()
    })
  })
})
