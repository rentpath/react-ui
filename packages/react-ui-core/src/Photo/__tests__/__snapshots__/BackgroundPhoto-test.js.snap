// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundPhoto fallbackUrl renders 1 url when not set 1`] = `
<div
  className=""
  data-tid="bg-photo"
  style={
    Object {
      "backgroundImage": "url(/foo)",
    }
  }
/>
`;

exports[`BackgroundPhoto fallbackUrl renders 2 urls when set 1`] = `
<div
  className=""
  data-tid="bg-photo"
  style={
    Object {
      "backgroundImage": "url(/foo), url(/bar)",
    }
  }
/>
`;
